gammapy/estimators/points/profile.py

```diff
@@ -133,12 +133,15 @@ def run(self, datasets):
         profile : `~gammapy.estimators.FluxPoints`
             Profile flux points.
         """
+
         datasets = Datasets(datasets=datasets)
+        datasets_copy = datasets.copy()
```

Comment:
Why is a copy needed. beyond the fact of an increased memory usage?

Reply:
> Why are initially models lost?

@bkhelifi I think this is our design, that dataset manipulation discards the models, which is the case here on calling `to_spectrum_dataset`.

Reply:
modified to avoid copy